### PR TITLE
Bug fix for data sanitization error for property value length, should…

### PR DESCRIPTION
**Pull request by jpiyali**
… be 8k

### CODE REVIEW

**Comment by jpiyali** [comment]
Issue Id is https://github.com/Microsoft/ApplicationInsights-JS/issues/566

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts [hunk 1]
```diff
@@ -67,15 +77,15 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
...
                 value = Util.trim(value);
-                if (value.toString().length > DataSanitizer.MAX_STRING_LENGTH) {
-                    value = value.toString().substring(0, DataSanitizer.MAX_STRING_LENGTH);
+                if (value.toString().length > maxLength) {
+                    value = value.toString().substring(0, maxLength);
                     _InternalLogging.throwInternal(
                         LoggingSeverity.WARNING,
                         _InternalMessageId.StringValueTooLong,
-                        "string value is too long. It has been truncated to " + DataSanitizer.MAX_STRING_LENGTH + " characters.",
+                        "string value is too long. It has been truncated to " + maxLength + " characters.",
```
Comment (gardnerjr) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: as a future task, can we possibly also get the *key* corresponding of the value that was trimmed in this message?  the message tells you what the *value* was that was truncated, but knowing that it was a custom property with name `x` or the field `url` or whatever would be really useful to track down what the source of that value was.

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts [hunk 2]
```diff
@@ -156,6 +166,23 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
             return measurements;
         }
 
+        public static sanitizeId(id: string) {
```
Comment (gardnerjr) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: you changed the other method to take a value for maxlength that has a default, why not do the same here?  (or why is there a different method for ids at all, when it's the same rules but different trim length?

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts [hunk 3]
```diff
@@ -156,6 +166,23 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
...
+            if (id) {
+                id = Util.trim(id);
+                if (id.length > DataSanitizer.MAX_ID_LENGTH) {
+                    id = id.substring(0, DataSanitizer.MAX_ID_LENGTH);
+                    _InternalLogging.throwInternal(
+                        LoggingSeverity.WARNING,
+                        _InternalMessageId.UrlTooLong,
+                        "id is too long, it has been truncated to " + DataSanitizer.MAX_URL_LENGTH + " characters.",
```
Comment (gardnerjr) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: the message has a different value (url length) than the checks did (id length)

Comment (gardnerjr) [reply JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: (and has messageid url too long)

File: JavaScript/JavaScriptSDK/Logging.ts [hunk 4]
```diff
@@ -86,6 +86,7 @@
         UrlTooLong,
         SessionStorageBufferFull,
         CannotAccessCookie,
+        IdTooLong
```
Comment (SergeyKanzhelev) [inline JavaScript/JavaScriptSDK/Logging.ts]: nit: it is a good practice to have comma in the end of the list.

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts (hunk 3 above)
Comment (jpiyali) [reply JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: Fixed.

File: JavaScript/JavaScriptSDK/Logging.ts (hunk 4 above)
Comment (jpiyali) [reply JavaScript/JavaScriptSDK/Logging.ts]: Updated.

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts (hunk 2 above)
Comment (jpiyali) [reply JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: Combined sanitizeUrl and sanitizeId() to call single sanitizeInput() method.

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts (hunk 1 above)
Comment (jpiyali) [reply JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: It is possible to do this now with the associated length (value is different for name vs. custom property). we use this method for these two types only).

File: JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts [hunk 5]
```diff
@@ -156,6 +154,27 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
             return measurements;
         }
 
+        public static sanitizeId(id: string): string {
+                return id ? DataSanitizer.sanitizeInput(id, DataSanitizer.MAX_ID_LENGTH, _InternalMessageId.IdTooLong).toString() : id;
+        }
+
+        public static sanitizeInput(input: any, maxLength: number, _msgId: _InternalMessageId) {
```
Comment (Dmitry-Matveev) [inline JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: `sanitizeInput()` is indirectly tested with `sanitizeUrl()` and `sanitizeId()` - would you like to explicitly test it as well?
It's in public API and one would not know that it's untested if one is to take dependency on this method later.

Comment (jpiyali) [reply JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts]: Added unit tests.

**Pull request merged**